Packs/IronPort/Integrations/CiscoEmailSecurityApplianceIronPortV2/CiscoEmailSecurityApplianceIronPortV2.py

```diff
@@ -0,0 +1,1928 @@
+from typing import Callable, Tuple
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+import uuid
+
+
+JWT_TOKEN_EXPIRATION_PERIOD = 30
+DEFAULT_FETCH = 50
+TIMESTAMP_FORMAT = "%d %b %Y %H:%M:%S (%Z +00:00)"
+QUARANTINE_TIMESTAMP_FORMAT = "%d %b %Y %H:%M (%Z +00:00)"
+
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+CISCO_TIME_FORMAT = "%Y-%m-%dT%H:%M:00.000Z"
+CISCO_REPORTING_TIME_FORMAT = "%Y-%m-%dT%H:00:00.000Z"
+
+MIN_PAGE_NUMBER = 1
+MIN_LIMIT = 1
+MIN_PAGE_SIZE = 1
+MAX_PAGE_SIZE = 100
+REQUEST_MAX_PULL = 100
+
+QUARANTINE_TYPE = "spam"
+VIEW_ACTION = "view"
+RELEASE_ACTION = "release"
+ADD_ACTION = "add"
+APPEND_ACTION = "append"
+EDIT_ACTION = "edit"
+
+
+class Client(BaseClient):
+    """Client class to interact with Cisco ESA API."""
+
+    def __init__(
+        self, server_url: str, username: str, password: str, verify: bool, proxy: bool
+    ):
+        super().__init__(base_url=server_url, headers={}, verify=verify, proxy=proxy)
+        self.username = username
+        self.password = password
+        self.handle_request_headers()
+
+    def handle_request_headers(self):
+        """Retrieve and save to integration context JWT token for authorized client class API requests."""
+        integration_context = get_integration_context()
+        jwt_token = integration_context.get("jwt_token")
+        jwt_token_issued_time = integration_context.get("jwt_token_issued_time")
+        if jwt_token and jwt_token_issued_time >= datetime.timestamp(
+            datetime.now() - timedelta(minutes=JWT_TOKEN_EXPIRATION_PERIOD)
+        ):
+            self._headers["jwtToken"] = jwt_token
+        else:
+            jwt_token = self.retrieve_jwt_token()
+            set_integration_context(
+                {"jwt_token": jwt_token, "jwt_token_issued_time": time.time()}
+            )
+            self._headers["jwtToken"] = jwt_token
+
+    def retrieve_jwt_token(self) -> str:
+        """
+        Retrieve JWT token from Cisco ESA.
+
+        Returns:
+            str: JWT token from Cisco ESA.
+        """
+        data = {
+            "data": {
+                "userName": b64_encode(self.username),
+                "passphrase": b64_encode(self.password),
+            }
+        }
+        try:
+            response = self._http_request("POST", "login", json_data=data)
```

Comment:
please use the `ok_codes` argument in `_http_request` to make sure the request returns a valid response (rather than 401, 403 etc), here as well as elsewhere in the integration. This way, invalid responses will raise an error (please make sure the error message is shown properly)